service/scaladsl/api/src/main/scala/com/lightbend/lagom/scaladsl/api/LagomConfigComponent.scala

```diff
@@ -0,0 +1,12 @@
+/*
+ * Copyright (C) 2016-2017 Lightbend Inc. <https://www.lightbend.com>
+ */
+package com.lightbend.lagom.scaladsl.api
+
+import com.typesafe.config.Config
+import play.api.Configuration
+
+trait LagomConfigComponent {
+  def configuration: Configuration
+  def config: Config = configuration.underlying
+}
```

Comment:
Because `config` is defined in terms of `configuration`, it won't be straightforward to remove the latter. It seems like we should be refactoring it to work the other way around.